libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/OAuthWebviewHelper.java

```diff
@@ -655,6 +655,13 @@ public void alias(String alias) {
 		try {
 			certChain = KeyChain.getCertificateChain(activity, alias);
 			key = KeyChain.getPrivateKey(activity, alias);
+			activity.runOnUiThread(new Runnable() {
+
+                @Override
+                public void run() {
```

Comment:
Loading the login page after the certificate has been loaded. Views can only be loaded from UI thread.